apps/emqx_durable_storage/src/emqx_ds_storage_layer.erl

```diff
@@ -256,12 +256,10 @@ make_iterator(
                     Err
             end;
         {error, not_found} ->
-            {error, end_of_stream}
+            {error, unrecoverable, generation_not_found}
```

Comment:
I think this error has to be split in two:

- Generation id is less than the first known generation: this should return something like `gone_permanently` (it's good to give it a name that user can understand).
- Generation id is _greater_ than the last known generation: this, perhaps, can be recoverable (it can happen due to replica being out of sync, or something like that)

Reply:
P.S. This can be fixed in a follow-up.

Reply:
Good point indeed.